SourceCode.py

```diff
@@ -1,35 +1,78 @@
 
 
 
+from ast import Pass
+from lib2to3.pgen2.token import OP
 from tkinter import Variable
+from unittest import case
 
 
 class Const : 
-    DATATYPE=["int","float","double","long"]
+    DATATYPE=["int","float","double","long","char"]
     OPERATIONS = ["+","-","*","/","++","--","*=","-=","/=","+="]
-    VARIABLES = []
+    VARIABLES = {}
 class Data:
-    def __init__(this,file_name):
-        this.data = open(file_name).read()
-        this.lines = this.data.split("\n")
-        this.tokens = {
-
-        }
-    def tokinizer(this):
-        for i in this.lines:
+    def __init__(self,file_name):
+        self.data = open(file_name).read()
+        self.lines = self.data.split("\n")
+        self.tokens = {}
+    def tokinizer(self):
+        for i in self.lines:
```

Comment:
what do mean by the name "i", please choose names that have meaning, ex. for **line** in lines 
and don't forget comments